dbcon/joblist/tuple-bps.cpp

```diff
@@ -273,7 +272,7 @@ TupleBPS::TupleBPS(const pColStep& rhs, const JobInfo& jobInfo) :
     fDelivery = false;
     fExtendedInfo = "TBPS: ";
     fQtc.stepParms().stepType = StepTeleStats::T_BPS;
-
+    fJoinMemLimit.reset(new int64_t(*jobInfo.umMemLimit == 0 ? fRm->getConfiguredUMMemLimit() / 4 : min((uint64_t)*jobInfo.umMemLimit, fRm->getConfiguredUMMemLimit() / 4)));
```

Comment:
What is the semantics of "/4" op ?

Reply:
The getMemory calls that use this value are determining if we should stop building a buffer but rather release the data and start a new buffer. I have no idea what is a good value to set here so I used 25% of totalUMMemory. I would love to hear your thoughts on this.
Note that we are still limited to totalUMMemory. fJoinMemLimit doesn't subtract from sessionLimit.

Before this patch, we didn't count this buffer memory, but rather filled until the loop finished: 99% of the time, this is OK, as the loop, for most JOINs, will be 8192 rows. But in the case of multiple matches, it can grow much larger. Hence the limiter.